### PR TITLE
Ports TG's instrument system and guitar

### PR DESCRIPTION
**Pull request by JustSumBody**
Continuing where Unid left off in #5442
To be honest he had most of the leg work done already, all that was really left was cleaning up the sound files and changing the violin/piano pathing in virtually every place it came up.
So we're moving to a new system that is flat based: rather than having duplicate sound files such as F#/Gb we just have the Gb file and the proc will send F# to Gb because music theory. It also eliminates reduntant files such as E# and directs them to Fn because why would you ever write an E# you animal.
~~Couldn't find the inhand sprite for the guitar from his original PR though, searching for icons is hard. I'll look a bit harder/bug someone else to make one if I can't find it.~~
Also of note: The guitar is limited in range. While the other instruments have a full range spanning nine octaves, the guitar is limited from the 3rd octave to the 6th or something like that I'm not that good at music. If you try to play something too high or low it just doesn't play.
"Tested" moderately
Sprite:
image

FOUND THE IN HANDS:
image
image
It's hard to get good ingame screens of the different views because of that whole layering thing, but you should get the idea
edit: Oh there it is, my /tg/ branch was outdated so I didn't see the guitar
now with video https://www.youtube.com/watch?v=HAWBF-G4CJM

### CODE REVIEW

**Comment by Exxion** [comment]
Thanks now I can make drunk people suck ass at playing music
Which is something I'm pretty sure I said I would do months ago
Oops

**Comment by ComicIronic** [comment]
>he doesn't write sharps
>he doesn't write DOUBLE sharps

Scope
this
pleb

File: code/game/objects/items/devices/instruments.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	qdel(song)
+	song = null
+	..()
+
+/obj/item/device/instrument/initialize()
+	song.tempo = song.sanitize_tempo(song.tempo) // tick_lag isn't set when the map is loaded
+	..()
+
+/obj/item/device/instrument/attack_self(mob/user as mob)
+	if(!ishuman(user) && !issilicon(user))
```
Comment (ComicIronic) [inline code/game/objects/items/devices/instruments.dm]: Use dexterity_check() instead.

File: code/datums/supplypacks.dm [hunk 2]
```diff
@@ -267,24 +267,25 @@ var/list/all_supply_groups = list("Supplies","Clothing","Security","Hospitality"
...
-	/obj/item/weapon/reagent_containers/glass/paint/violet,
-	/obj/item/weapon/reagent_containers/glass/paint/black,
-	/obj/item/weapon/reagent_containers/glass/paint/white,
-	/obj/item/weapon/reagent_containers/glass/paint/remover,
-	/obj/item/mounted/poster,
-	/obj/item/stack/package_wrap/gift,
-	/obj/item/stack/package_wrap/gift,
-	/obj/item/stack/package_wrap/gift)
-	cost = 10
+					/obj/item/device/camera,
```
Comment (ComicIronic) [inline code/datums/supplypacks.dm]: You should avoid making formatting changes for non-repo-standard reasons.

Comment (JustSumBody) [inline code/datums/supplypacks.dm]: It was brought up in the original PR, but that was also like a year ago.
Should I undo it?

Comment (ComicIronic) [inline code/datums/supplypacks.dm]: Yes.

**Comment by JustSumBody** [comment]
Actually wait the inhands disappeared somehow

**Comment by JustSumBody** [comment]
~~Why is it saying the file changed but not actually adding the icons what the fuck is this
Am I really being stopped by this of all things~~
Maybe I am special, it's in ready 4 merge

File: code/__HELPERS/text.dm [hunk 3]
```diff
@@ -471,6 +471,26 @@ var/list/number_units=list(
 			i++
 	//testing(" ([recursion]) out=list("+jointext(out,", ")+")")
 	return out
-
+	
+//Function from Byond's "TextHandling" Library, given a string turns it into a list separated by a given character
+/proc/dd_text2List(text, separator)
```
Comment (RemieRichards) [inline code/__HELPERS/text.dm]: why not just use `splittext()`?

Comment (JustSumBody) [inline code/__HELPERS/text.dm]: Because I didn't know that existed.

**Pull request merged**